genienlp/models/general_seq2seq.py

```diff
@@ -199,14 +277,98 @@ def generate(self,
         return generated
         
 
-    
-            
+           
+class Bart(torch.nn.Module):
```

Comment:
Shouldn't Bart inherit from Seq2Seq? They share similar methods with different "forward" and perhaps "generation" function. In a sense, Seq2Seq will be our "base model" which all future models inherit from.

Reply:
Is there any advantage in inheriting from Seq2Seq? There won't be any shared code, and methods don't work the same way. I agree with Sina here, it should not inherit.

Reply:
We will add new models with encoder-decoder architecture and would be nice if they share a similar interface not necessarily Seq2Seq (similar to huggingface having PreTrainedModel as the base model). We can postpone it to when we actually have mode models.

Reply:
The only "interface" they need is the constructor and the "forward" method. Everything else is an implementation detail of each model. Now, I imagine Bart is actually going to be used for any Huggingface model, but that's separate.

Reply:
We will have two classes: our Seq2Seq (which is BERT-LSTM and friends) and a TrasformerSeq2Seq (which is BART, T5, etc.). We will have to wait till we update the `transformers` version to 3.5, and then we can change this Bart class to a more general TransformerSeq2Seq.